flax/jax_utils.py

```diff
@@ -117,7 +117,8 @@ def partial_eval_by_shape(fn, input_spec, *args, **kwargs):
   if config.omnistaging_enabled:
     _, out_pvals, _ = pe.trace_to_jaxpr(f_flat, in_pvals)
   else:
-    _, out_pvals, _ = pe.trace_to_jaxpr(f_flat, in_pvals, stage_out=True)
+    with jax.core.initial_style_staging():
```

Comment:
thanks for adding this, we'll also do a review of all recent merges into master's old "nn" lib and rebase onto master again as well to make sure we don't miss any important commits.

Reply:
Sounds good!